modules/data_release/ajax/FileUpload.php

```diff
@@ -28,59 +47,69 @@
 
     $baseURL = $settings->getBaseURL();
 
-    if (!file_exists(__DIR__ . "/../user_uploads/")) {
-        error_log(
-            "ERROR: File upload failed. Default user_uploads"
-            . " directory not found."
+    $target_path = $base_path . $fileName;
+    if (move_uploaded_file($_FILES["file"]["tmp_name"], $target_path)) {
+        // insert the file into the data_release table
+        $DB->insert(
+            'data_release',
+            array(
+             'file_name'   => $fileName,
+             'version'     => $version,
+             'upload_date' => $upload_date,
+            )
         );
-        header("HTTP/1.1 500 Internal Server Error");
-    } elseif (!is_writable(__DIR__ . "/../user_uploads/")) {
-        error_log(
-            "File upload failed. Default user_uploads directory"
-            . " does not appear to be writeable."
+        // get the ID of the user who uploaded the file
+        $user_ID = $DB->pselectOne(
+            "SELECT ID FROM users WHERE userid=:UserID",
+            array('UserID' => $user->getUsername())
         );
-        header("HTTP/1.1 500 Internal Server Error");
-    } else {
-        $target_path = $base_path . $fileName;
-        if (move_uploaded_file($_FILES["file"]["tmp_name"], $target_path)) {
-            $DB->insert(
-                'data_release',
-                array(
-                 'file_name'   => $fileName,
-                 'version'     => $version,
-                 'upload_date' => $upload_date,
-                )
-            );
+        // get the ID of the file inserted in the data_release table
+        $version_where = $version ? "version=:version" : "version IS :version";
+        $ID            = $DB->pselectOne(
+            "SELECT 
+               id 
+             FROM 
+               data_release 
+             WHERE 
+               file_name=:file_name 
+               AND $version_where
+               AND upload_date=:upload_date",
+            array(
+             'file_name'   => $fileName,
+             'version'     => $version,
+             'upload_date' => $upload_date,
+            )
+        );
+        // add permission to the user for the uploaded data_release file
+        $DB->insert(
+           'data_release_permissions',
+            array(
+             'userid'          => $user_ID,
+             'data_release_id' => $ID,
+            )
+        );
+    }
+    header("Location: {$baseURL}/data_release/?uploadSuccess=true");
+    header("201 Created");
```

Comment:
```suggestion
    header("HTTP/1.1 201 Created");
```

Alternatively, `http_response_code(201)`